mobile/src/notifications/notification.epics.ts

```diff
@@ -28,40 +29,50 @@ const notificationsHandler$ = (action$: ActionsObservable<LoadUserFinished>, sta
                 Push.setListener({
                     onPushNotificationReceived: notification => {
 
-                        const userInfo = state$.value.userInfo;
-                        if (!userInfo || !userInfo.employeeId) {
-                            return;
-                        }
-
-                        const isAndroid = Platform.OS === 'android';
-                        if (isAndroid && !!notification.message) {
+                        if (Platform.OS === 'android' && !!notification.message) {
                             // Android messages received in the background don't include a message. On Android, that fact can be used to
                             // check if the message was received in the background or foreground. For iOS the message is always present.
                             return;
                         }
 
-                        if (!notification.customProperties) {
+                        if (!state$.value.userInfo || !state$.value.userInfo.employeeId ||
+                            !notification.customProperties || !notification.customProperties.type) {
                             return;
                         }
 
-                        const employeeId = notification.customProperties.employeeId;
-                        const approverId = notification.customProperties.approverId;
-
-                        observer.next(loadPendingRequests());
+                        const currentEmployeeId = state$.value.userInfo.employeeId;
+                        const { type, employeeId, approverId, url } = notification.customProperties;
 
-                        if (employeeId) {
-                            observer.next(loadCalendarEvents(employeeId));
-                            if (userInfo.employeeId === employeeId) {
-                                observer.next(openProfile());
-                            } else {
-                                observer.next(openEmployeeDetails(employeeId));
-                            }
-                        } else if (approverId) {
-                            if (userInfo.employeeId === approverId) {
-                                observer.next(openProfile());
-                            }
-                        } else {
-                            logError('Unexpected notification payload', notification.customProperties);
+                        switch (type) {
+                            case NotificationType.UpdateAvailable:
+                                if (!url) {
+                                    logError('Unexpected notification payload', notification.customProperties);
+                                } else {
+                                    Linking.openURL(url);
```

Comment:
Add // noinspection JSIgnoredPromiseFromCall above

Reply:
add a catch instead? So it doesn't crash for weird reason? `Linking.openUrl(url).catch( console.error )`

Reply:
It can crash only in case of none-http links, but anyway it's really better to handle it instead of ignoring. c6ada5463594568dcb2f3b2ce5e75d4087bb7766